.github/workflows/build.yaml

```diff
@@ -0,0 +1,102 @@
+name: Auto Build ETCD RPM
+
+on:
+  push:
+    branches: ["main", "release/**"]
+    tags:
+      - "**"
+    paths-ignore:
+      - "docs/**"
+  pull_request:
+    types: [labeled, synchronize]
+    branches: ["release/**", "main", "dev/**"]
+    paths-ignore:
+      - "docs/**"
+
+concurrency:
+  group: auto-build-etcd-rpm-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  auto_build_etcd_rpm:
+    if: ${{ contains(github.event.pull_request.labels.*.name, 'ci-build') ||
+      contains(github.event.pull_request.labels.*.name, 'ci-all') ||
+      github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/tags/') }}
+    name: auto build etcd rpm package
+    runs-on: ubuntu-20.04
+    timeout-minutes: 40
+
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+
+      - name: setup go
+        uses: actions/setup-go@v2.1.4
+        with:
+          go-version: "1.19"
+
+      - name: install nfpm
+        run: |
+          go install github.com/goreleaser/nfpm/v2/cmd/nfpm@latest
+
+      - name: Extract etcd version
+        id: etcd_version
+        shell: bash
+        run: |
+          echo "##[set-output name=version;]$(cat ./VERSION)"
+
+      - name: download etcd
+        run: |
+          export VERSION=${{ steps.etcd_version.outputs.version }}
+          wget https://github.com/etcd-io/etcd/releases/download/v${VERSION}/etcd-v${VERSION}-linux-amd64.tar.gz
+          tar xzvf etcd-v${VERSION}-linux-amd64.tar.gz
+          # mv etcd and etcdctl to current folder
+          mv etcd-v${VERSION}-linux-amd64/etcd etcd-v${VERSION}-linux-amd64/etcdctl .
+
+      - name: Build rpm package
+        run: |
+          export VERSION=${{ steps.etcd_version.outputs.version }}
+
+          # build etcd rpm package
+          version=${VERSION} nfpm pkg -p rpm -f nfpm.yaml -t etcd-${VERSION}.x86_64.rpm
+
+      - name: Install podman
+        run: |
+          sudo apt update
+          sudo apt -y install podman
+
+      - name: Install rpm package in centos7 container
+        run: |
+          export VERSION=${{ steps.etcd_version.outputs.version }}
+          sudo podman run -d --name centos7Instance --systemd true --privileged --net host centos:7 /usr/lib/systemd/systemd
+          sudo podman cp etcd-${VERSION}.x86_64.rpm centos7Instance:/etcd-${VERSION}.x86_64.rpm
+          sudo podman exec centos7Instance yum -y install /etcd-${VERSION}.x86_64.rpm
+          sudo podman exec centos7Instance systemctl enable etcd
+          sudo podman exec centos7Instance systemctl start etcd
+
+      - name: run test case
+        run: |
+          export VERSION=${{ steps.etcd_version.outputs.version }}
+          ./scripts/etcd_test.sh ${VERSION}
+
+      - name: Publish Artifact
+        uses: actions/upload-artifact@v2.2.3
+        with:
+          name: "etcd-rpm"
+          path: "./etcd-${{ steps.etcd_version.outputs.version }}.x86_64.rpm"
```

Comment:
maybe we could upload it to the release

Reply:
i remove this step and create a new step to upload rpm to release.

<img width="1270" alt="image" src="https://user-images.githubusercontent.com/28869910/233243508-478e2c62-1fea-4fa6-a56a-d78e0d4ef989.png">